spec/lib/tasks/dev_rake_spec.rb

```diff
@@ -330,7 +330,7 @@
       stub_request_token
       stub_request_enroll
 
-      expect_any_instance_of(Object).to receive(:sleep).exactly(10).times.with(0.2)
```

Comment:
oh I didn't even see we had `expect_any_instance_of` in here! Yeah no wonder this spec was flaking :shakes-fist:!

Reply:
For my own curiosity, is it generally bad practice to use `expect_any_instance_of` in rspec?  Or is it problematic when it's used in this specific way with `sleep`?  (I think it's the latter, but want to check I'm not missing anything.)

Reply:
personally I think that `expect_any_instance_of` or `allow_any_instance_of` are kind of antipatterns. when stubbing, I think it's crucial to know which instance you're stubbing on, since there can be a lot of different things happening (like with a common method like `sleep`)

However, the `_any_instance_of` methods are useful in situations where we don't have access to the instances (acceptance specs, or classes that don't expose clear methods to stub), or cases like this where we don't know what instance the rake task is being called on.

And I think the issue here is that `sleep` is a fairly common method, and it's mixed in to every object via `Kernel` so it was just an unforunate combination of things that lead to this flakiness.

Reply:
Got it, thank you.